lib/util.js

```diff
@@ -147,11 +149,48 @@ function styleText(format, text, { validateStream = true, stream = process.stdou
       validateOneOf(key, 'format', ObjectKeys(inspect.colors));
     }
     if (skipColorize) continue;
-    left += escapeStyleCode(formatCodes[0]);
-    right = `${escapeStyleCode(formatCodes[1])}${right}`;
+    ArrayPrototypePush(codes, formatCodes);
   }
 
-  return skipColorize ? text : `${left}${text}${right}`;
+  if (skipColorize) {
+    return text;
+  }
+
+  // Build opening codes
+  let openCodes = '';
+  for (let i = 0; i < codes.length; i++) {
+    openCodes += escapeStyleCode(codes[i][0]);
+  }
+
+  // Process the text to handle nested styles
+  let processedText;
+  if (codes.length > 0) {
+    processedText = ArrayPrototypeReduce(
+      codes,
+      (text, code) => RegExpPrototypeSymbolReplace(
+        new RegExp(`\\u001b\\[${code[1]}m`, 'g'),
+        text,
+        (match, offset) => {
+          // Check if there's more content after this reset
+          if (offset + match.length < text.length) {
+            return `${match}${escapeStyleCode(code[0])}`;
```

Comment:
this seems cause extra chars in the text

Reply:
does this need to actually drop the original?

this looks like it'll replace `x1b[22m` with `x1b[22mx1b[2m` for example

but it probably wants to _replace_ the match? so `x1b[22m` becomes `x1b[2m`

Reply:
```
assert.strictEqual(
  util.styleText('dim',
    'dim' +
      util.styleText('bold', 'bold') +
    'dim',
  '\x1B[2mdim\x1b[1mbold\x1b[22m\x1b[2mdim\x1b[22m'
);
```

in this case, I cannot think up a way to handle the logic.

because they have the same closing tag. So the only way is to check whether the opening tag is in the text

Reply:
The way picocolors does it is by searching for the closing sequence, and replacing it with a close/open 

But in all other cases, it replaces with an open (removes the close) 

So `{39m}foo {38m}bar{30m} baz{30m}` (without nesting support) would become `{39m}foo {38m}bar{39m} baz{30m}`

But `{2m}dim {1m}bold{22m} dim{22m}` would become `{2m}dim {1m}bold{22m}{2m} dim{22m}`